### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by alexcrichton**
Just a usual `cargo update` plus moving over some major versions

### CODE REVIEW

File: src/cargo/util/errors.rs [hunk 1]
```diff
@@ -86,7 +86,8 @@ impl CargoError {
             &CargoErrorKind::Parse(_) |
             &CargoErrorKind::Git(_) |
             &CargoErrorKind::Internal(_) |
-            &CargoErrorKind::CargoTestErrorKind(_) => false
+            &CargoErrorKind::CargoTestErrorKind(_) |
+            &CargoErrorKind::__Nonexhaustive { .. } => false
```
Comment (matklad) [inline src/cargo/util/errors.rs]: This seems like something is wrong somewhere...

Should we ask that error chain has an option not to generate Nonexhaustive?

Error chain issue: https://github.com/rust-lang-nursery/error-chain/issues/182

Comment (alexcrichton) [reply src/cargo/util/errors.rs]: Is this wrong though? It seems like it's fine to match "nonexhaustive errors" inside of Cargo itself?

Comment (matklad) [reply src/cargo/util/errors.rs]: Hm, yeah, kinda buy "all consumers must add `_`, but we can use __Nonexaustive".

**Pull request merged**